src/index.ts

```diff
@@ -240,6 +249,10 @@ class logs {
     extension: string | null,
     msgs: any
   ): string => {
+    if (typeof this._formatFunc === "function") {
+      return this._formatFunc(level, extension, msgs);
+    }
```

Comment:
This is a great addition, and something I'd been looking for! I do wonder though whether the date formatting should happen before this fork so that the date can also be passed into `_formatFunc` rather than having to replicate that logic in the customer `_formatFunc` implementation?